lib/smart_proxy_ansible/plugin.rb

```diff
@@ -19,6 +19,8 @@ class Plugin < Proxy::Plugin
         require 'smart_proxy_ansible/runner/command_creator'
         require 'smart_proxy_ansible/runner/playbook'
 
+        raise NotExistingWorkingDirException.new("Working Directory does not exist") unless Proxy::Ansible::Plugin.settings[:working_dir].nil? || File.directory?(Proxy::Ansible::Plugin.settings[:working_dir])
```

Comment:
It should also expand the path. IIRC the rest of the code can deal with paths like `~/.foreman-ansible` just fine, this one does not.